### PR TITLE
readline: bump version

### PR DESCRIPTION
**Pull request by kenmays**

### CODE REVIEW

File: sys-libs/readline/readline-8.2.013.recipe [hunk 1]
```diff
@@ -10,20 +10,33 @@ The history facilites are also placed into a separate library, the History \
...
 Readline in applications which desire its capabilities."
 HOMEPAGE="http://tiswww.case.edu/php/chet/readline/rltop.html"
-COPYRIGHT="1989-2019 Free Software Foundation, Inc."
+COPYRIGHT="1989-2024 Free Software Foundation, Inc."
 LICENSE="GNU GPL v3"
 REVISION="1"
 SOURCE_URI="ftp://ftp.gnu.org/gnu/readline/readline-8.2.tar.gz"
 SOURCE_DIR="readline-8.2"
-for i in 001; do
+for i in 013; do
```
Comment (jmairboeck) [inline sys-libs/readline/readline-8.2.013.recipe:18]: This requires a sequence, like in the deleted recipes:

```suggestion
for i in {001..013}; do
```

Your code would only use patch number 013, not the other ones before.

File: sys-libs/readline/readline-8.2.013.recipe [hunk 2]
```diff
@@ -64,7 +77,7 @@ defineDebugInfoPackage readline$secondaryArchSuffix \
 
 PATCH()
 {
-	for i in 001; do
+	for i in 013; do
```
Comment (jmairboeck) [inline sys-libs/readline/readline-8.2.013.recipe:80]: the same here

**Review by jmairboeck — CHANGES_REQUESTED** [review]

**Comment by jmairboeck** [comment]
Have you checked whether the older versions are still used anywhere before removing them?

**Comment by kenmays** [comment]
Yes.

**Comment by Begasus** [comment]
> Yes.

Doesn't seem so, trying to uninstall readline7 here also removes gnuchess and gnugo.

**Comment by Begasus** [comment]
Keep older readline recipes around, please don't go removing packages that are potentional in use, readline6 and readline7 are NOT the same as readline8, so they don't superseed eachother or conflict eachother.

**Comment by kenmays** [comment]
hrev58057 x86:
```
/Desktop> pkgman uninstall readline7*
The following changes will be made:
  in system:
    uninstall package readline7_x86_devel-7.0.5-1
    uninstall package readline7_x86-7.0.5-1
Continue? [yes/no] (yes) : Continue? [yes/no] (yes) : 
[system] Applying changes ...
[system] Changes applied. Old activation state backed up in "state_2024-09-06_03:42:56"
[system] Cleaning up ...
[system] Done.
```

Why not migrate those legacy Haikuports apps to readline8 for R1B5 ? Build system currently defaults to readline-8.2.001-1-x86_gcc2.hpkg.

**Comment by pulkomandy** [comment]
We can always keep migrating stuff, but then we'll never get the release shipped. This can just as well be done after the release.

In any case, if you are interested in doing that migration, you need to rebuild each affected recipe, ideally update their build_prerequires to make sure they use the new version, and only when we have checked that nothing uses the older readline versions, we can remove them. Here you start by removing the old version, so this is doing it the wrong way around.

**Comment by kenmays** [comment]
Ok.. adding them back....

**Comment by Begasus** [comment]
> hrev58057 x86:
> 
> ```
> /Desktop> pkgman uninstall readline7*
> The following changes will be made:
>   in system:
>     uninstall package readline7_x86_devel-7.0.5-1
>     uninstall package readline7_x86-7.0.5-1
> Continue? [yes/no] (yes) : Continue? [yes/no] (yes) : 
> [system] Applying changes ...
> [system] Changes applied. Old activation state backed up in "state_2024-09-06_03:42:56"
> [system] Cleaning up ...
> [system] Done.
> ```
> 
> Why not migrate those legacy Haikuports apps to readline8 for R1B5 ? Build system currently defaults to readline-8.2.001-1-x86_gcc2.hpkg.

That doesn't mean nothing uses it, it just tells you that on "your" install it's not used by anything.

**Comment by kenmays** [comment]
hrev58057 x86:
```
~/haikuports/games-board/gnuchess> grep readline *.recipe
        lib:libreadline$secondaryArchSuffix
        devel:libreadline$secondaryArchSuffix
~/haikuports/games-board/gnuchess> cd ../..
~/haikuports> cd */gnugo
~/haikuports/games-board/gnugo> grep readline *.recipe
        lib:libreadline
        devel:libreadline
~/haikuports/games-board/gnugo> pkgman install gnugo
  100% repochecksum-1 [65 bytes]
Validating checksum for Haiku...done.
  100% repochecksum-1 [64 bytes]
Validating checksum for HaikuPorts...done.
  100% repochecksum-1 [71 bytes]
Validating checksum for LOTE...done.
The following changes will be made:
  in system:
    install package gnugo-3.8-1 from repository HaikuPorts
Continue? [yes/no] (yes) : 
  100% gnugo-3.8-1-x86_gcc2.hpkg [1.89 MiB]
Validating checksum for https://eu.hpkg.haiku-os.org/haikuports/master/x86_gcc2/current/packages/gnugo-3.8-1-x86_gcc2.hpkg...done.
[system] Applying changes ...
[system] Changes applied. Old activation state backed up in "state_2024-09-06_03:53:51"
[system] Cleaning up ...
[system] Done.

~/haikuports/games-board/gnugo> pkgman install gnuchess_x86
  100% repochecksum-1 [65 bytes]
Validating checksum for Haiku...done.
  100% repochecksum-1 [64 bytes]
Validating checksum for HaikuPorts...done.
  100% repochecksum-1 [71 bytes]
Validating checksum for LOTE...done.
The following changes will be made:
  in system:
    install package gnuchess_x86-6.2.7-1 from repository HaikuPorts
Continue? [yes/no] (yes) : 
  100% gnuchess_x86-6.2.7-1-x86_gcc2.hpkg [714.65 KiB]
Validating checksum for https://eu.hpkg.haiku-os.org/haikuports/master/x86_gcc2/current/packages/gnuchess_x86-6.2.7-1-x86_gcc2.hpkg...done.
[system] Applying changes ...
[system] Changes applied. Old activation state backed up in "state_2024-09-06_04:36:16"
[system] Cleaning up ...
[system] Done.

~/haikuports/games-board/gnugo> ls /system/packages/readline*
/system/packages/readline_x86-8.2.001-1-x86_gcc2.hpkg
/system/packages/readline-8.2.001-1-x86_gcc2.hpkg
~/haikuports/games-board/gnugo> 

```

No dependency requirements for readline6/readline7....

**Comment by Begasus** [comment]
```
~> readelf -d /bin/gnuchess

Dynamic section at offset 0x59b70 contains 25 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [libintl.so.8]
 0x0000000000000001 (NEEDED)             Shared library: [libreadline.so.7]
 0x0000000000000001 (NEEDED)             Shared library: [libncurses.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libstdc++.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libgcc_s.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libroot.so]
```

**Comment by kenmays** [comment]
hrev58057 x86:
~/haikuports/games-board/gnugo> readelf -d  /bin/gnuchess
```
Dynamic section at offset 0x58fe8 contains 25 entries:
  Tag        Type                         Name/Value
 0x00000001 (NEEDED)                     Shared library: [libintl.so.8]
 0x00000001 (NEEDED)                     Shared library: [libreadline.so.8]
 0x00000001 (NEEDED)                     Shared library: [libncurses.so.6]
 0x00000001 (NEEDED)                     Shared library: [libstdc++.so.6]
 0x00000001 (NEEDED)                     Shared library: [libgcc_s.so.1]
 0x00000001 (NEEDED)                     Shared library: [libroot.so]
```
This is from the installed gnuchess_x86 from Haikuports today.

**Comment by Begasus** [comment]
@jmairboeck LGTM

**Review by jmairboeck — APPROVED** [review]

**Pull request merged**